code/addons/test/src/node/vitest-manager.ts

```diff
@@ -83,36 +71,49 @@ export class VitestManager {
     });
   }
 
-  async runTests(testPayload: TestingModuleRunRequestPayload['payload']) {
+  async runTests(requestPayload: TestingModuleRunRequestPayload) {
     if (!this.vitest) {
       await this.startVitest();
     }
-
-    // This list contains all the test files (story files) that need to be run
-    // based on the test files that are passed in the tests array
-    // This list does NOT contain any filtering of specific
-    // test cases (story) within the test files
-    const testList: TestSpecification[] = [];
-
-    const storybookTests = await this.getStorybookTestSpecs();
-
-    for (const storybookTest of storybookTests) {
-      const match = testPayload.find((test) => {
-        const absoluteImportPath = path.join(process.cwd(), test.importPath);
-        return absoluteImportPath === storybookTest.moduleId;
-      });
-      if (match) {
-        // make sure to clear the file cache so test results are updated even if watch mode is not enabled
-        if (!this.testManager.watchMode) {
-          this.updateLastChanged(storybookTest.moduleId);
+    this.resetTestNamePattern();
+
+    const index = (await fetch(requestPayload.indexUrl).then((res) => res.json())) as StoryIndex;
+    const { storyIds = Object.keys(index.entries) } = requestPayload;
+    const stories = storyIds.map((id) => index.entries[id]);
```

Comment:
Extract this part into a private method.

Reply:
Done 👍